### PR TITLE
Introduce upgrade recovery image

### PR DESCRIPTION
**Pull request by donpenney**
This update introduces the upgrade recovery image, which is
responsible for writing the upgrade recovery script to the recovery
partition and taking the pre-upgrade backup.

Co-Authored-By: Daniel Mellado <dmellado@redhat.com>
Co-Authored-By: Yolanda Robla <yroblamo@redhat.com>
Co-Authored-By: Don Penney <dpenney@redhat.com>

Signed-off-by: Don Penney <dpenney@redhat.com>

### CODE REVIEW

**Comment by donpenney** [comment]
/hold
This PR depends on merge of https://github.com/openshift-kni/cluster-group-upgrades-operator/pull/125

**Comment by donpenney** [comment]
Note that the current test failures are due to the missing dependency being introduced by #125 ... Once that PR merges, these test should be able to pass.

File: recovery/cmd/launchBackup.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright 2021 Red Hat, Inc.
```
Comment (irinamihai) [inline recovery/cmd/launchBackup.go:2]: 2022?

Comment (donpenney) [reply recovery/cmd/launchBackup.go:2]: Code was started last year in a dev repo. I meant to check for dates... I'll fix these up now

File: recovery/cmd/root.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+}
+
+func init() {
+	cobra.OnInitialize(initConfig)
+
+	// Here you will define your flags and configuration settings.
+	// Cobra supports persistent flags, which, if defined here,
+	// will be global for your application.
+
+	rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.openshift-ai-image-backup.yaml)")
```
Comment (sabbir-47) [inline recovery/cmd/root.go:59]: I'm not sure if we need this persistent flag.

File: recovery/cmd/root.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+}
+
+// initConfig reads in config file and ENV variables if set.
+func initConfig() {
```
Comment (sabbir-47) [inline recovery/cmd/root.go:64]: Also, i think we are not reading anything from config file or passing any environment variable, so i'm not sure if we need it

**Review by sabbir-47 — COMMENTED** [review]
Please have a look at 2 comments that i've added

File: recovery/cmd/root.go (hunk 3 above)
Comment (donpenney) [reply recovery/cmd/root.go:64]: I wondered about these when I was integrating the code, but didn't want to modify too much. Once #125 merges, though, I will update this PR with this code dropped.

File: recovery/cmd/root.go (hunk 2 above)
Comment (donpenney) [reply recovery/cmd/root.go]: Removed

File: recovery/cmd/root.go (hunk 3 above)
Comment (donpenney) [reply recovery/cmd/root.go]: Removed

**Review by sabbir-47 — COMMENTED** [review]
/lgtm

**Pull request merged**